File: c++/src/leetcode/_I_1610_M.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int maxAliveYear(vector<int>& birth, vector<int>& death) {
        vector<int> changes(102, 0);
        for (int i = 0, size = birth.size(); i < size; ++i) {
            changes[birth[i] - 1900]++;
            changes[death[i] - 1900 + 1]--;
        }
        int curAlive = 0;
        int maxAlive = 0;
        int maxAliveYear = 1900;
        for (int i = 0, size = changes.size(); i < size; ++i) {
            if ((curAlive += changes[i]) > maxAlive) {
                maxAlive = curAlive;
                maxAliveYear = 1900 + i;
            }
        }
        return maxAliveYear;
    }
};

int main() {
    Solution solution;
    vector<int> b {1900, 1901, 1950};
    vector<int> d {1948, 1951, 2000};
    cout << solution.maxAliveYear(b, d) << endl;
}

File: python/leetcode/_I_1610_M.py
from typing import List


class Solution:
    def maxAliveYear(self, birth: List[int], death: List[int]) -> int:
        num = len(birth)
        changes = [0] * 102
        for i in range(num):
            changes[birth[i] - 1900] += 1
            changes[death[i] - 1900 + 1] -= 1
        curAlive = 0
        maxAlive = 0
        maxAliveYear = 1900
        for i, change in enumerate(changes):
            curAlive += change
            if maxAlive < curAlive:
                maxAlive = curAlive
                maxAliveYear = 1900 + i
        return maxAliveYear


if __name__ == '__main__':
    print(Solution().maxAliveYear([1900, 1901, 1950], [1948, 1951, 2000]))
